aip/general/0193.md

```diff
@@ -472,41 +392,49 @@ variable information, so there needs to be machine-readable component of
-differs from `Status.message` include the following:
-
-- The service requires an end-user-facing error message that differs
-  from the "debug message".
-- Ongoing, iterative error message improvements are expected.
+error messages. While `LocalizedMessage` **may** use a locale specified
+in the request, a service **may** provide a `LocalizedMessage` even without
+a user-specified locale, typically to provide a better error message in
+[situations where `Status.message` cannot be changed](updating-statusmessage).
+Where the locale is not specified by the user, it **should** be `en-US`
+(US English).
+
+A service **may** include `LocalizedMessage` even when the same message is
+provided in `Status.message` and when localization into a user-specified locale
+is not supported. Reasons for this include:
+
+- An intention to support user-specified localization in the near future, allowing
+  clients to consistently use `LocalizedMessage` and not change their error-reporting
+  code when the functionality is introduced.
+- Consistency across all RPCs within a service: if some RPCs include
+  `LocalizedMessage` and some only use `Status.message` for error messages, clients
+  have to be aware of which RPCs will do what, or implement a fall-back mechanism.
+  Providing `LocalizedMessage` on all RPCs allows simple and consistent client code
+  to be written.
 
 ### Updating Status.message
 
-**Because Status.message is part of the API contract**, avoid updating
-it entirely in favor of updating LocalizedMessage; for example, clients
-often perform string matches on the text to differentiate one error for
-another and even parse the error message to extract variables from
-dynamic segments. 
-
-Clients **must** only perform string matches if their only option for
-extracting dynamic information is the message itself. If structured data
-is present in metadata, the recommended practice is to key into that
-data instead.
+If a client has ever observed an error with `Status.message` populated
+(which it always will be) but without `ErrorInfo`, the developer of that client
+may well have had to resort to parsing `Status.message` in order to find out
+information beyond just what `Status.code` conveys. That information may be
+found by matching specific text (e.g. "Connection closed with unknown cause")
+or by parsing the message to find out metadata values (e.g. a region with
+insufficient resources). At that point, `Status.message` is implicitly part
+of the API contract, so **must not** be updated - that would be a breaking
+change. This is one reason for introducing `LocalizedMessage` into the
+`Status.details`.
+
+RPCs which have **always** included `ErrorInfo` are in a better position:
+the contract is then more about the stability of `ErrorInfo` for any given
+error: the reason and domain need to be consistent over time, and the
+metadata provided for any given (reason,domain) can only be expanded.
```

Comment:
No, it's horrible - sorry about that. I think changing the second colon to a period would be better. Will do that locally, but not create a commit just for that. (I'll wait until we get more feedback to avoid a storm of tiny commits.)